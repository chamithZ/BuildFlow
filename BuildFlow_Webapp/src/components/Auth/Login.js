import React, { useState } from "react";
import { useNavigate } from "react-router-dom"; // Import useNavigate
import {
    Flex,
    Box,
    FormControl,
    FormLabel,
    Input,
    Checkbox,
    Stack,
    Button,
    Heading,
    Text,
    useColorModeValue,
} from "@chakra-ui/react";
import { LoginSupplier } from "../../Services/AuthServices";
import Swal from "sweetalert2";
import "./ClientLogin.css";

const Login = () => {
    const navigate = useNavigate(); // Define the navigate function

    const [formData, setFormData] = useState({
        email: "",
        password: "",
    });

    const { email, password } = formData;

    const onChange = (e) =>
        setFormData({ ...formData, [e.target.name]: e.target.value });

    const onSubmit = async (e) => {
        e.preventDefault();
        let data = await LoginSupplier(formData);
        if (data.data.token) {
            localStorage.setItem("token", data.data.token);
            localStorage.setItem("userRole", data.data.userRole);
            localStorage.setItem("user", data.data.user);
            localStorage.setItem("userId", data.data.userId);
            Swal.fire(
                "Congrats",
                "Successfully Login to your Account",
                "success"
            );
            navigate("/Profile");
        } else {
            Swal.fire("error", "Try Again!", "error");
            navigate("/Profile");
        }
    };

    return (
        <Flex
            minH={"100vh"}
            align={"center"}
            justify={"center"}
            bg={useColorModeValue("gray.50", "gray.800")}
        >
            <Stack spacing={8} mx={"auto"} maxW={"lg"} py={12} px={6}>
                <Stack align={"center"}>
                    <Heading fontSize={"4xl"}>Sign in to your account</Heading>
                    <Text fontSize={"lg"} color={"gray.600"}>
                        to enjoy all of our cool{" "}
                        <Text color={"blue.400"}>features</Text> ✌️
                    </Text>
                </Stack>
                <Box
                    rounded={"lg"}
                    bg={useColorModeValue("white", "gray.700")}
                    boxShadow={"lg"}
                    p={8}
                >
                    <form onSubmit={onSubmit}>
                        <Stack spacing={4}>
                            <FormControl id="email">
                                <FormLabel>Email address</FormLabel>
                                <Input
                                    type="email"
                                    name="email"
                                    value={email}
                                    onChange={onChange}
                                />
                            </FormControl>
                            <FormControl id="password">
                                <FormLabel>Password</FormLabel>
                                <Input
                                    type="password"
                                    name="password"
                                    minLength="6"
                                    value={password}
                                    onChange={onChange}
                                />
                            </FormControl>
                            <Stack spacing={10}>
                                <Stack
                                    direction={{ base: "column", sm: "row" }}
                                    align={"start"}
                                    justify={"space-between"}
                                >
                                    <Checkbox>Remember me</Checkbox>
                                    <Text color={"blue.400"}>
                                        Forgot password?
                                    </Text>
                                </Stack>
                                <Button
                                    type="submit"
                                    bg={"blue.400"}
                                    color={"white"}
                                    _hover={{
                                        bg: "blue.500",
                                    }}
                                >
                                    Sign in
                                </Button>
                            </Stack>
                        </Stack>
                    </form>
                </Box>
            </Stack>
        </Flex>
    );
};

export default Login;
