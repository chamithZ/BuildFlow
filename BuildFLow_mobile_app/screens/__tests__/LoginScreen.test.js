/* eslint-disable prettier/prettier */
import React from 'react';
import renderer from 'react-test-renderer';
import Adapter from 'enzyme-adapter-react-16'
import {shallow, configure} from 'enzyme'
import Login from '../LoginScreen';

configure({ adapter: new Adapter() });

it("renders correctly", () => {
    shallow(<Login />);
});

it('email check', () => {
    const login = shallow(<Login />)
    login.find('TextInput[keyboardType="email-address"]').simulate('change', {
      nativeEvent: {
        text: 'gasballa@gmail.com',
      },
    });
    expect(login.find('TextInput[keyboardType="email-address"]').prop('value')).toEqual(
      'gasballa@gmail.com',
    );
});

it('password Check', () => {
    const login = shallow(<Login />)
    login.find('TextInput[secureTextEntry=true]').simulate('change', {
      nativeEvent: {
        text: 'somenewpassword',
      },
    });
    expect(login.find('TextInput[secureTextEntry=true]').prop('value')).toEqual(
      'somenewpassword',
    );
});



