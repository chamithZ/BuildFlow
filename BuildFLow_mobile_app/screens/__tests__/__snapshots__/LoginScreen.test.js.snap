// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login component renders correctly 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "backgroundColor": "white",
      "flex": 1,
      "justifyContent": "center",
      "width": "100%",
    }
  }
>
  <Image
    resizeMode="contain"
    source={
      {
        "testUri": "../../../assets/Images/LoginBackground.png",
      }
    }
    style={
      {
        "height": "40%",
        "width": "100%",
      }
    }
  />
  <RCTScrollView
    style={
      {
        "width": "80%",
      }
    }
  >
    <View>
      <TextInput
        keyboardType="email-address"
        onChange={[Function]}
        placeholder="E-mail Address"
        style={
          {
            "alignItems": "center",
            "backgroundColor": "white",
            "borderColor": "#00716F",
            "borderRadius": 12,
            "borderWidth": 2,
            "color": "#FF9720",
            "fontSize": 16,
            "marginHorizontal": 24,
            "marginTop": 10,
            "marginVertical": 4,
            "paddingHorizontal": 16,
            "paddingVertical": 10,
            "textAlign": "center",
          }
        }
        value=""
      />
      <TextInput
        onChange={[Function]}
        placeholder="Password"
        secureTextEntry={true}
        style={
          {
            "alignItems": "center",
            "backgroundColor": "white",
            "borderColor": "#00716F",
            "borderRadius": 12,
            "borderWidth": 2,
            "color": "#FF9720",
            "fontSize": 16,
            "marginHorizontal": 24,
            "marginTop": 10,
            "marginVertical": 4,
            "paddingHorizontal": 16,
            "paddingVertical": 10,
            "textAlign": "center",
          }
        }
        value=""
      />
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "alignItems": "center",
            "backgroundColor": "#FF9720",
            "borderRadius": 23,
            "justifyContent": "center",
            "marginHorizontal": 16,
            "marginVertical": 20,
            "opacity": 1,
            "paddingHorizontal": 4,
            "paddingVertical": 10,
          }
        }
      >
        <Text
          style={
            {
              "color": "white",
              "fontSize": 16,
            }
          }
        >
          Login
        </Text>
      </View>
    </View>
  </RCTScrollView>
</View>
`;
